zebra-state/src/service/non_finalized_state/chain.rs

```diff
@@ -663,6 +677,22 @@ impl Chain {
             .map(|(_height, tree)| tree.clone())
     }
 
+    /// Returns the Sapling [`NoteCommitmentSubtree`](sapling::tree::NoteCommitmentSubtree) specified
+    /// by a [`HashOrHeight`], if it exists in the non-finalized [`Chain`].
+    pub fn sapling_subtree(
```

Comment:
Will we have subtrees in the non-finalized state? Given their width at the leaf level, we would have at most one at a time.

Reply:
> Will we have subtrees in the non-finalized state?

Only for finalizing after they reach the root of the non-finalized state.